fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AvailableDeparturesPage.js

```diff
@@ -133,8 +131,14 @@ class CDUAvailableDeparturesPage {
                     // Clicking the already selected runway is a no-op
                     if (!selected) {
                         mcdu.onLeftInput[i + 1] = async () => {
+                            // TODO: cleanup hotfix solution
                             try {
                                 await mcdu.flightPlanService.setOriginRunway(runway.ident, forPlan, inAlternate);
+                                let _targetPlan = mcdu.flightPlan(forPlan, inAlternate);
+                                if (selectedSid && !_targetPlan.availableDepartures.some(sid => sid.databaseId === selectedSid.databaseId)) {
+                                    await mcdu.flightPlanService.setDepartureProcedure(null, forPlan, inAlternate);
+                                    await mcdu.flightPlanService.setDepartureEnrouteTransition(null, forPlan, inAlternate);
```

Comment:
This type of logic should be in the flight planner rather than in the CDU pages.

Reply:
Fair, this was just to patch the problem as quickly as possible